src/main/kotlin/tanvd/grazi/ide/ui/GraziAddDeleteListPanel.kt

```diff
@@ -1,21 +1,46 @@
 package tanvd.grazi.ide.ui
 
-import com.intellij.openapi.ui.ComboBox
-import com.intellij.openapi.ui.DialogBuilder
-import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.actionSystem.ActionToolbarPosition
+import com.intellij.openapi.ui.JBPopupMenu
 import com.intellij.ui.AddDeleteListPanel
-import com.intellij.ui.components.JBLabel
+import com.intellij.ui.CommonActionsPanel
+import com.intellij.ui.ListUtil
+import com.intellij.ui.ToolbarDecorator
 import com.intellij.util.ui.JBUI
 import tanvd.grazi.GraziConfig
 import tanvd.grazi.language.Lang
+import java.awt.BorderLayout
 import java.awt.Component
+import java.awt.event.ActionEvent
 import javax.swing.*
 
-class GraziAddDeleteListPanel : AddDeleteListPanel<Lang>(null, GraziConfig.get().enabledLanguages.sortedWith(Comparator.comparing(Lang::displayName))) {
-    private val cbLanguage = ComboBox<Lang>()
+interface GraziLanguagePanelUpdateListener {
```

Comment:
Why do you need interface with only two functions? You may just have two constructor parameters in `GraziAddDeleteListPanel` -- lambda `onLanguageAdded` and lambda `onLanguageRemoved`

Reply:
Done